server/controllers/ctoToolController.js

```diff
@@ -11,39 +15,109 @@ async function handleCTOToolUse({
+
     const searchResults = await searchService.performSearch(searchQuery);
+
+    // Extract images from search results and remove trailing slashes
+    const images = (searchResults.images || []).map((url) =>
+      url.replace(/\/$/, "")
+    );
+
+    let messageContent = [
+      {
+        type: "text",
+        text: `Here are relevant images found for the query "${searchQuery}". Use these images as inspiration or placeholders when designing and creating components for the website:`,
+      },
+    ];
+
+    if (images.length > 0) {
+      // Add images to the message content
+      for (const imageUrl of images) {
+        try {
+          // Fetch the image
+          const response = await axios.get(imageUrl, {
+            responseType: "arraybuffer",
+          });
+
+          // Get the Content-Type from the response headers
+          const contentType = response.headers["content-type"];
+
+          // Only process if it's an image
+          if (contentType && contentType.startsWith("image/")) {
+            const imageData = Buffer.from(response.data).toString("base64");
+
+            messageContent.push({
+              type: "image",
+              source: {
+                type: "base64",
+                media_type: contentType,
+                data: imageData,
+              },
+            });
+          } else {
+            console.error(
+              `Unable to determine media type for image ${imageUrl}`
+            );
+          }
+        } catch (error) {
+          console.error(`Error processing image ${imageUrl}:`, error);
+        }
+      }
+    }
+
     return [
       {
         type: "tool_result",
         tool_use_id: tool.id,
-        content: [{ type: "text", text: JSON.stringify(searchResults) }],
+        content: messageContent,
       },
     ];
-  } 
-  else if (tool.name === TOOLS.IMAGE_FINDER) {
+  } else if (tool.name === TOOLS.IMAGE_FINDER) {
     const searchQuery = tool.input.query;
-    console.log("Performing image search with query:", searchQuery);
+
     const imageResults = await searchService.imageSearch(searchQuery);
-    console.log("imageResults:", imageResults);
+
     return [
       {
         type: "tool_result",
         tool_use_id: tool.id,
         content: [
           {
             type: "text",
-            text: imageResults.length === 0 ? "No relevant images found" : JSON.stringify(imageResults)
-          }
+            text:
+              imageResults.length === 0
+                ? "No relevant images found"
+                : JSON.stringify(imageResults),
+          },
         ],
       },
     ];
-  }
-  else if (tool.name === TOOLS.FILE_CREATOR) {
+  } else if (tool.name === TOOLS.IMAGE_ANALYSIS) {
+    const { image_url, analysis_prompt } = tool.input;
+
+    const analysisResults = await analyzeImage(image_url, analysis_prompt);
+
+    // Process the analysis results
+    const processedResults = {
```

Comment:
dont go with this, let ai decide

Reply:
I will work on this while building analyze and reapir tool, need to decide on the structure of the analysis and how we can use it